node/rpc/src/lib.rs

```diff
@@ -51,114 +72,331 @@ where
+        .with_method("Filecoin.StateMinerInfo", state_miner_info::<DB, KS>, false)
+        .with_method(
+            "Filecoin.StateMinerFaults",
+            state_miner_faults::<DB, KS>,
+            false,
         )
-        .with_method("Filecoin.StateMinerInfo", state_miner_info::<DB, KS>)
-        .with_method("Filecoin.StateMinerFaults", state_miner_faults::<DB, KS>)
         .with_method(
             "Filecoin.StateAllMinerFaults",
             state_all_miner_faults::<DB, KS>,
+            false,
         )
         .with_method(
             "Filecoin.StateMinerRecoveries",
             state_miner_recoveries::<DB, KS>,
+            false,
+        )
+        .with_method("Filecoin.StateReplay", state_replay::<DB, KS>, false)
+        .with_method("Filecoin.StateGetActor", state_get_actor::<DB, KS>, false)
+        .with_method(
+            "Filecoin.StateAccountKey",
+            state_account_key::<DB, KS>,
+            false,
         )
-        .with_method("Filecoin.StateReplay", state_replay::<DB, KS>)
-        .with_method("Filecoin.StateGetActor", state_get_actor::<DB, KS>)
-        .with_method("Filecoin.StateAccountKey", state_account_key::<DB, KS>)
-        .with_method("Filecoin.StateLookupId", state_lookup_id::<DB, KS>)
+        .with_method("Filecoin.StateLookupId", state_lookup_id::<DB, KS>, false)
         .with_method(
             "Filecoin.StateMartketBalance",
             state_market_balance::<DB, KS>,
+            false,
         )
-        .with_method("Filecoin.StateGetReceipt", state_get_receipt::<DB, KS>)
-        .with_method("Filecoin.StateWaitMsg", state_wait_msg::<DB, KS>)
+        .with_method(
+            "Filecoin.StateGetReceipt",
+            state_get_receipt::<DB, KS>,
+            false,
+        )
+        .with_method("Filecoin.StateWaitMsg", state_wait_msg::<DB, KS>, false)
         // Gas API
         .with_method(
             "Filecoin.GasEstimateGasLimit",
             gas_estimate_gas_limit::<DB, KS>,
+            false,
         )
         .with_method(
             "Filecoin.GasEstimateGasPremium",
             gas_estimate_gas_premium::<DB, KS>,
+            false,
+        )
+        .with_method(
+            "Filecoin.GasEstimateFeeCap",
+            gas_estimate_fee_cap::<DB, KS>,
+            false,
         )
-        .with_method("Filecoin.GasEstimateFeeCap", gas_estimate_fee_cap::<DB, KS>)
         .finish_unwrapped();
 
-    let mut app = tide::Server::with_state(rpc);
-    app.at("/rpc/v0").post(handle_json_rpc);
-    app.listen(rpc_endpoint).await.unwrap();
+    let try_socket = TcpListener::bind(rpc_endpoint).await;
+    let listener = try_socket.expect("Failed to bind to addr");
+    let state = Arc::new(rpc);
+
+    info!("waiting for web socket connections");
+    while let Ok((stream, addr)) = listener.accept().await {
+        task::spawn(handle_connection_and_log(
+            state.clone(),
+            stream,
+            addr,
+            subscriber.clone(),
```

Comment:
be careful cloning the subscribers, this would probably change if you change the previous comments, but the subscriber that's being cloned isn't being polled off